instrumented/sink.go

```diff
@@ -57,22 +58,34 @@ func (ams *AsyncMessageSink) PublishMessages(ctx context.Context, acks chan<- su
 			case <-ctx.Done():
 				return <-errs
 			case err := <-errs:
-				if err != nil {
+				if isUnexpectedError(err) {
 					ams.counter.WithLabelValues("error", ams.topic).Inc()
 				}
 				return err
 			}
 		case <-ctx.Done():
 			return <-errs
 		case err := <-errs:
-			if err != nil {
+			if isUnexpectedError(err) {
 				ams.counter.WithLabelValues("error", ams.topic).Inc()
 			}
 			return err
 		}
 	}
 }
 
+func isUnexpectedError(err error) bool {
+	if err == nil {
+		return false
+	}
+	// we're expecting producers to mark the stopping of producing by cancelling the context
+	if errors.Is(err, context.Canceled) {
+		return false
+	}
+
+	return true
```

Comment:
Nit: I would maybe go for a switch statement here instead of the two ifs, but that's just me
```suggestion
	switch {
	case err == nil:
		return false
	case errors.Is(err, context.Canceled):
		// we're expecting producers to mark the stopping of producing by cancelling the context
		return false
	default:
		return true
	}
```
actually this could just be, but that's probably not as clear
```
return err != nil && !errors.Is(err, context.Canceled)
```

Reply:
The ```return err != nil && !errors.Is(err, context.Canceled)``` I had in the first place, and was thinking to go with the switch myself for clarity. Thanks for the suggestion.